knowledge-base/pivotgrid-local-date-aggregates.md

```diff
@@ -0,0 +1,169 @@
+---
+title: Use Date Aggregates with Local Data
+description: Learn how to implement and configure aggregate calculations by date periods like year, month, week, and day, when using local data binding with the Telerik PivotGrid for Blazor.
+type: how-to
+page_title: How to Use Date Aggregates with Local Data
+slug: pivotgrid-kb-local-date-aggregates
+tags: blazor, pivotgrid, aggregates
+ticketid: 1678109
+res_type: kb
+---
+
+## Environment
+
+<table>
+    <tbody>
+        <tr>
+            <td>Product</td>
+            <td>PivotGrid for Blazor</td>
+        </tr>
+    </tbody>
+</table>
+
+## Description
+
+This KB answers the following questions:
+
+* How to allow users to aggregate local data by date periods, such as years, months, weeks, and days?
+* How to enable date aggregations with local PivotGrid data binding?
+* How to calculate date aggregates by period with a local data provider?
+
+## Solution
+
+The [PivotGrid supports date aggregates by period](slug:pivotgrid-data-binding#usage-differences) only with [XMLA data binding](slug:pivotgrid-data-binding#xmla). Consider the following approach for local data scenarios:
+
+1. Define all `DateTime` properties in the PivotGrid model class as `internal` or `private`, so that the PivotGrid ignores them.
```

Comment:
```suggestion
1. Define all `DateTime` properties in the PivotGrid model class as `internal`, `protected`, or `private`, so that the PivotGrid ignores them.
```